toolchain/check/impl_lookup.cpp

```diff
@@ -125,9 +126,24 @@ static auto GetWitnessIdForImpl(Context& context, SemIR::LocId loc_id,
     return SemIR::InstId::None;
   }
 
-  // TODO: If the interface id of the `impl` and the query are not the same,
-  // then we can skip this `impl`. (The interface id is the root of the
-  // constraint, the unique `interface` declaration.)
+  auto facet_type = context.types().GetAs<SemIR::FacetType>(
+      context.types().GetTypeIdForTypeConstantId(interface_const_id));
+  auto facet_type_info = context.facet_types().Get(facet_type.facet_type_id);
+  if (facet_type_info.impls_constraints.size() != 1) {
+    context.TODO(loc_id,
+                 "impl lookup for a FacetType with more than one interface");
+    return SemIR::InstId::None;
+  }
```

Comment:
We're calling `GetWitnessIdForImpl` in a loop, and `interface_const_id` is invariant throughout that loop. Maybe we could move this into the caller and pass in the `interface_id`?

Reply:
Done, also added a separate TODO when there's no constraints, which shows up in one place now